src/libsystemd/sd-netlink/netlink-message-nfnl.c

```diff
@@ -192,12 +197,33 @@ int sd_nfnl_call_batch(
         if (r < 0)
                 return r;
 
-        for (size_t i = 0; i < n_messages; i++)
-                RET_GATHER(r, sd_netlink_read(nfnl, serials[i], usec, /* ret= */ NULL));
-        if (r < 0)
-                return r;
+        for (size_t i = 1; i <= n_messages; i++)
+                /* If we have received an error, kernel may not send replies for later messages. Let's ignore
+                 * remaining replies. */
+                if (r < 0)
+                        (void) sd_netlink_ignore_serial(nfnl, serials[i], usec);
```

Comment:
maybe do `continue` here so that there's no rason to indent the reset of the loop body?